cast/src/lib.rs

```diff
@@ -1236,6 +1277,38 @@ impl SimpleCast {
     }
 }
 
+pub struct RawLogs<'a, 'b, M: Middleware> {
+    provider: &'b M,
+    block_stream: FilterWatcher<'a, <M as Middleware>::Provider, TxHash>,
+    filter: Filter,
+}
+impl<'a, 'b, M: Middleware> Unpin for RawLogs<'a, 'b, M> {}
+
+impl<'a, 'b, M: Middleware> Stream for RawLogs<'a, 'b, M>
+where
+    M: Middleware,
+{
+    type Item = Result<Vec<Log>, <M as Middleware>::Error>;
+    fn poll_next(
+        mut self: Pin<&mut Self>,
+        cx: &mut TaskContext<'_>,
+    ) -> Poll<Option<Result<Vec<Log>, <M as Middleware>::Error>>> {
+        match Pin::new(&mut self.block_stream).poll_next(cx) {
+            Poll::Ready(block) => {
+                println!("New block! {:?}", block);
+                // Apply the new filter so that it shows logs from the latest block
+                let filter = self.filter.clone().at_block_hash(block.unwrap());
```

Comment:
ideally the stream should yield the block so that we don't need to unwrap here

Reply:
What do you mean? It's an option I need to unwrap so that I can update the log's filter with the current block